quiche/src/lib.rs

```diff
@@ -6453,17 +6460,21 @@ impl Connection {
                         p.recovery.delivery_rate_update_app_limited(true);
                     }
 
-                    let (lost_packets, lost_bytes) = p.recovery.on_ack_received(
-                        &ranges,
-                        ack_delay,
-                        epoch,
-                        handshake_status,
-                        now,
-                        &self.trace_id,
-                    )?;
+                    let (lost_packets, lost_bytes, acked_bytes) =
```

Comment:
The counter is increased by the number of stream data bytes buffered, but this counts _all_ bytes of a packet (including other frames, packet headers, ...) so it's always going to decrease the counter by more bytes than were actually buffered.

Instead I think the counter needs to be decreased only when STREAM frames specifically are acked, and only decrease by the actual amount of stream data that was acked, see https://github.com/cloudflare/quiche/blob/master/quiche/src/lib.rs#L2689

Reply:
I think for DATAGRAMs it's easier because we already have the total queue size, but yeah, I guess the check in `delivery_rate_check_if_app_limited()` should use something like `self.tx_buffered + self.dgram_send_queue.byte_size()`.

Reply:
thanks will address these things.